xla/python/weakref_lru_cache.cc

```diff
@@ -240,10 +236,13 @@ class WeakrefLRUCache : public std::enable_shared_from_this<WeakrefLRUCache> {
         inserted = true;
         return std::make_shared<CacheEntry>();
       });
+      ++total_queries_;
+      if (inserted) {
```

Comment:
You could make this

```
if (inserted || !entry->completed.HasBeenNotified())
```

which slightly overcounts (if the entry is in the process of being populated) but I think seems better to me.